packages/firestore/src/core/sync_engine.ts

```diff
@@ -912,7 +914,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
       const activeTargets: TargetId[] = [];
 
       let p = Promise.resolve();
-      objUtils.forEachNumber(this.queriesByTarget, (targetId, _) => {
+      this.queriesByTarget.forEach((_, targetId) => {
```

Comment:
Could these `forEach` loops be replaced with simpler and, arguable, more readable,`for` loops?

e.g.

```
this.queriesByTarget.forEach((_, targetId) => {
```

could be changed to:


```
for (targetId of this.queriesByTarget.keys()) {
```

Reply:
I tried making this work (I pushed two commits and one revert). The problem is that I would have to turn on "downlevelIteration" to make this work with ES5, which adds a lot of code (+4%).

The example you posted actually does not need downlevelIteration since you are converting the Map to an Array. That creates an extra copy though and hence I decided to leave this as is.

Reply:
Cool.  Thanks for looking into this.